pkg/operator/staticpod/controller/node/node_controller.go

```diff
@@ -72,9 +77,27 @@ func (c NodeController) sync(ctx context.Context, syncCtx factory.SyncContext) e
 			}
 		}
 		if found {
-			newTargetNodeStates = append(newTargetNodeStates, originalOperatorStatus.NodeStatuses[i])
+			newTargetNodeState := applyoperatorv1.NodeStatus().
+				WithNodeName(originalOperatorStatus.NodeStatuses[i].NodeName).
+				WithCurrentRevision(originalOperatorStatus.NodeStatuses[i].CurrentRevision).
+				WithTargetRevision(originalOperatorStatus.NodeStatuses[i].TargetRevision).
+				WithLastFailedRevision(originalOperatorStatus.NodeStatuses[i].LastFailedRevision).
+				WithLastFailedReason(originalOperatorStatus.NodeStatuses[i].LastFailedReason).
+				WithLastFailedCount(originalOperatorStatus.NodeStatuses[i].LastFailedCount).
+				WithLastFallbackCount(originalOperatorStatus.NodeStatuses[i].LastFallbackCount).
+				WithLastFailedRevisionErrors(originalOperatorStatus.NodeStatuses[i].LastFailedRevisionErrors...)
+			if originalOperatorStatus.NodeStatuses[i].LastFailedTime != nil {
+				newTargetNodeState.WithLastFailedTime(*originalOperatorStatus.NodeStatuses[i].LastFailedTime)
+			}
+			newTargetNodeStates = append(newTargetNodeStates, newTargetNodeState)
 		} else {
 			syncCtx.Recorder().Warningf("MasterNodeRemoved", "Observed removal of master node %s", nodeState.NodeName)
+			jsonPatch.WithRemove(fmt.Sprintf("/status/nodeStatuses/%d", i), jsonpatch.NewTestCondition(fmt.Sprintf("/status/nodeStatuses/%d/nodeName", i), nodeState.NodeName))
+		}
+	}
+	if !jsonPatch.IsEmpty() {
+		if err = c.operatorClient.PatchOperatorStatus(ctx, jsonPatch); err != nil {
+			return err
```

Comment:
this will never show in degraded.  need to write status  in order to get the degraded condition.

Reply:
> this will never show in degraded. need to write status in order to get the degraded condition.

And when you do this, remember to write the node status we do know about in this state.

Reply:
every `err` should be reported as `degraded` condition ?